src/components/navigation/navigation-link/navigation-link.directive.ts

```diff
@@ -0,0 +1,139 @@
+import { LocationStrategy } from '@angular/common';
+import { Directive, HostBinding, HostListener, Input, OnChanges, OnDestroy, OnInit } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { filter, takeUntil, delay } from 'rxjs/operators';
+import { Subject } from 'rxjs/Subject';
+import { NavigationItem } from '../navigation-item.inferface';
+import { NavigationService } from '../navigation.service';
+
+@Directive({
+    selector: '[uxNavigationLink]',
+    exportAs: 'uxNavigationLink'
+})
+export class NavigationLinkDirective implements OnInit, OnChanges, OnDestroy {
+
+    @Input()
+    navigationItem: NavigationItem;
+
+    @Input()
+    set expanded(value: boolean) {
+        this._expanded$.next(value);
+    }
+
+    @Input()
+    canExpand: boolean;
+
+    @Input()
+    @HostBinding('class.indent')
+    indent: boolean;
+
+    @HostBinding('attr.href')
+    href: string;
+
+    @HostBinding('attr.role')
+    role: string;
+
+    @HostBinding('attr.aria-expanded')
+    ariaExpanded: string = 'undefined';
+
+    isActive: boolean;
+
+    indentChildren: boolean;
+
+    private _expanded$ = new Subject<boolean>();
+    private _onDestroy = new Subject<void>();
+
+    constructor(
+        private _router: Router,
+        private _locationStrategy: LocationStrategy,
+        private _navigationService: NavigationService
+    ) { }
+
+    ngOnInit(): void {
+
+        this._expanded$.pipe(delay(0), takeUntil(this._onDestroy)).subscribe(expanded => {
+            if (this.navigationItem.children && this.navigationItem.children.length > 0) {
```

Comment:
This may not be an issue, but I notice when autocollapsing a different section there is a bit of jumpiness happening.

https://i.gyazo.com/405042b0109659b55083a03b09e302ea.mp4

This seems to be because there is a slight delay - I assume this is to prevent an expression changed after check error. I notice that using `requestAnimationFrame` for the delay seems to resolve this, eg:
```typescript
this._expanded$.pipe(takeUntil(this._onDestroy)).subscribe(expanded => {
            requestAnimationFrame(() => {
                if (this.navigationItem.children && this.navigationItem.children.length > 0) {
                    this.ariaExpanded = `${expanded}`;
                    this._navigationService.setExpanded(this.navigationItem, expanded);
                }
            });
        });
`

This may be fine as is, but just thought i'd mention it.